pkg/registry/controller/etcd/etcd.go

```diff
@@ -45,7 +45,7 @@ func NewREST(h tools.EtcdHelper) *REST {
 		NewListFunc: func() runtime.Object { return &api.ReplicationControllerList{} },
 		// Produces a path that etcd understands, to the root of the resource
 		// by combining the namespace in the context with the given prefix
-		KeyRootFunc: func(ctx api.Context) string {
+		KeyRootFunc: func(ctx api.Context) (string, error) {
```

Comment:
I get that you want to check this at every layer, and I think at one point before we had generic.etcd the shared KeyRootFunc did throw an error, and @lavalamp removed it in the migration because it was kind of tedious to code against.  Note to self to check how tedious this ends-up being as I continue to review.